packages/ilmomasiina-frontend/src/routes/AuditLog/AuditLogActionFilter.tsx

```diff
@@ -2,27 +2,32 @@ import React from 'react';
 
 import { Form } from 'react-bootstrap';
 
+import { AuditEvent } from '@tietokilta/ilmomasiina-models/src/enum';
 import { setAuditLogQueryField } from '../../modules/auditLog/actions';
 import { useTypedDispatch } from '../../store/reducers';
 
 const ACTIONS = [
-  ['event.create', 'Tapahtuma: Luo'],
-  ['event.edit', 'Tapahtuma: Muokkaa'],
-  ['event.publish', 'Tapahtuma: Julkaise'],
-  ['event.unpublish', 'Tapahtuma: Luonnokseksi'],
-  ['event.delete', 'Tapahtuma: Poista'],
-  ['signup.edit', 'Ilmoittautuminen: Muokkaa'],
-  ['signup.delete', 'Ilmoittautuminen: Poista'],
-  ['signup.queuePromote', 'Ilmoittautuminen: Nousi jonosta'],
-  ['user.create', 'Käyttäjä: Luo'],
-  ['user.delete', 'Käyttäjä: Poista'],
+  [AuditEvent.CREATE_EVENT, 'Tapahtuma: Luo'],
+  [AuditEvent.EDIT_EVENT, 'Tapahtuma: Muokkaa'],
+  [AuditEvent.PUBLISH_EVENT, 'Tapahtuma: Julkaise'],
+  [AuditEvent.UNPUBLISH_EVENT, 'Tapahtuma: Luonnokseksi'],
+  [AuditEvent.DELETE_EVENT, 'Tapahtuma: Poista'],
+  [AuditEvent.EDIT_SIGNUP, 'Ilmoittautuminen: Muokkaa'],
+  [AuditEvent.DELETE_SIGNUP, 'Ilmoittautuminen: Poista'],
+  [AuditEvent.PROMOTE_SIGNUP, 'Ilmoittautuminen: Nousi jonosta'],
+  [AuditEvent.CREATE_USER, 'Käyttäjä: Luo'],
+  [AuditEvent.DELETE_USER, 'Käyttäjä: Poista'],
 ];
 
 const AuditLogActionFilter = () => {
   const dispatch = useTypedDispatch();
 
   const onChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
-    dispatch(setAuditLogQueryField('action', e.target.value));
+    const event = e.target.value;
+    dispatch(setAuditLogQueryField(
+      'action',
+      (Object.values(AuditEvent) as string[]).includes(event) ? event as unknown as [AuditEvent] : undefined,
+    ));
```

Comment:
Doesn't crash if the cast is erroneous, so keep it simple. (This suggestion also needs the changes to `src/schema/auditLog/index.ts` and `src/routes/auditlog/getAuditLogs.ts`)
```suggestion
    const event = e.target.value as AuditEvent;
    dispatch(setAuditLogQueryField('action', event));
```

Reply:
Resolved in 1f8f0ee